core/src/test/scala/integration/kafka/api/TransactionsExpirationTest.scala

```diff
@@ -97,6 +111,9 @@ class TransactionsExpirationTest extends KafkaServerTestHarness {
     producer.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic1, 3, "3", "3", willBeCommitted = true))
     producer.commitTransaction()
 
+    val describeState2 = admin.describeTransactions(Collections.singletonList("transactionalProducer")).description("transactionalProducer")
+    TestUtils.waitUntilTrue(() => describeState2.isDone, "Transactional state was never added.")
```

Comment:
nit: Would it make sense to factor out those two lines into a helper method to avoid the repetition?

Reply:
I didn't know if it was really that much better to make a helper if it was only used 2 times. but if we think its worth it we can.